staging/src/k8s.io/apiserver/pkg/registry/rest/update.go

```diff
@@ -126,7 +126,12 @@ func BeforeUpdate(strategy RESTUpdateStrategy, ctx context.Context, obj, old run
 	if err != nil {
 		return err
 	}
-	objectMeta.SetGeneration(oldMeta.GetGeneration())
+	if len(oldMeta.GetAnnotations()[genericapirequest.AnnotationKey]) == 0 || objectMeta.GetGeneration() == 0 {
```

Comment:
I think that comparing to an empty string is more readable but I can change to `_, found := accessor.GetAnnotations()[genericapirequest.AnnotationKey]; found`